base/complex.jl

```diff
@@ -363,13 +361,40 @@ function /(z::ComplexF64, w::ComplexF64)
     absa = abs(a); absb = abs(b);  ab = absa >= absb ? absa : absb # equiv. to max(abs(a),abs(b)) but without NaN-handling (faster)
     absc = abs(c); absd = abs(d);  cd = absc >= absd ? absc : absd
 
-    # constants
-    ov = floatmax(Float64)
-    un = floatmin(Float64)
-    ϵ  = eps(Float64)
-    halfov = 0.5*ov
-    twounϵ = un*2.0/ϵ
-    bs = 2.0/(ϵ*ϵ)
+    halfov = 0.5*floatmax(Float64)              # overflow threshold
+    twounϵ = floatmin(Float64)*2.0/eps(Float64) # underflow threshold
+
+    # actual division operations
+    if  ab>=halfov || ab<=twounϵ || cd>=halfov || cd<=twounϵ # over/underflow case
+        p,q = scaling_cdiv(a,b,c,d,ab,cd) # scales a,b,c,d before division (unscales after)
+    else
+        p,q = cdiv(a,b,c,d)
+    end
+
+    return ComplexF64(p,q)
+end
+
+# sub-functionality for /(z::ComplexF64, w::ComplexF64)
+function cdiv(a::Float64, b::Float64, c::Float64, d::Float64)
```

Comment:
This doesn't seem to get inlined

Reply:
If I check with `@code_typed` I'm only seeing invoked-calls to `robust_cdiv1` and `scaling_cdiv` (but no calls to `cdiv`)- I assume that means they're the only non-inlined calls? 
I guess it might be preferable to have `robust_cdiv1` inlined though.

Reply:
Worth benchmarking, but presumably you'd want the main body with a non-inlined slow case for normalization and then the normalization body with everything inlined into it.